indicators/NO_GLAC_001/R/NO_GLAC_001.qmd

```diff
@@ -0,0 +1,1731 @@
+  labs(y = "km-2")
+ggsave("../img/bcreg.png")
+```
+
+```{r fig-sonerOgSeksjoner}
+#| fig-cap: "Distribution of area within each combination of bioclimatic sone and section (i.e. each bioclimatic region)."
+
+knitr::include_graphics("../img/bcreg.png")
+```
+
+```{r}
+#| eval: false
+#| code-fold: true
+#| code-summary: "Union and cache bioclimatic region data"
+
+BCreg <- BCreg |>
+  group_by(BCregion) |>
+  summarize(Shape = st_union(Shape))
+saveRDS(BCreg, paste0(path, "/data/bioclimatic_regions.rds"))
+```
+
+```{r}
+#| include: false
+# read cache
+BCreg <- readRDS(paste0(path, "/data/bioclimatic_regions.rds"))
+```
+
+
+
+### 7.5 Dataset E - Alpine ecosystem extent
+
+I use a map of alpine extent that was also used by @framstad2021. It is
+based on certain open, terrestrial land cover types above the current
+/realised forest line. There is ongoing work to replace the ecosystem
+mask for the alpine main ecosystem. When such a map is ready, the
+workflow here can easily be updated.
+
+```{r}
+#| eval: false
+#| code-fold: true
+#| code-summary: "Prepare and cache alpine extent map"
+
+pathE <- paste0(path, "/data/temp/alpine_extent.tif")
+
+alpine <- stars::read_stars(pathE) |>
+  st_as_sf(merge = T)
+
+# intersections are much quicker on multipolygons, so I will union the geometries. 
+alpine <- st_union(alpine)
+
+saveRDS(alpine, paste0(path, "/data/alpine_extent.rds"))
+```
+
+```{r}
+#| include: false
+alpine <- readRDS(paste0(path, "/data/alpine_extent.rds"))
+```
+
+## 8. Spatial units
+
+I use basically three spatial units: EAs (level 1; 70 units),
+geographical region (level 2, 5 units), and the EAA.
+
+### 8.1 Ecosystem assets (EA)
+
+Here I create a dataset for the EAs. The EAs will be defined as the
+intersection between bioclimatic regions and geographical regions. Then
+I add total mountain area and glacier area as variables to this dataset.
+
+Note that the EA are not contiguous, and so metric values shold not be 
+presented at the EA level as maps since data from other geographical areas inside 
+each EA strata is informed from the entire EA.
+
+```{r}
+#| eval: false
+#| code-summary: 'Create EA dataset'
+EA <- BCreg |>
+  sf::st_intersection(greg)
+
+# create new variable for the unique EA IDs
+EA <- EA |>
+  mutate(EA_ID = row_number())
+
+saveRDS(EA, paste0(path, "/data/EAs.rds"))
+```
+
+```{r}
+#| include: false
+EA <- readRDS(paste0(path, "/data/EAs.rds"))
+```
+
+```{r mountainArea}
+#| warning: false
+#| code-summary: 'Intersect EAs and alpine extent map'
+
+# get mountain area for each EA 
+temp_alpine <- EA |>
+  st_intersection(alpine) |>
+  mutate(mountainArea = units::drop_units(st_area(Shape))) |>
+  as_tibble()
+
+#temp_alpine |>
+#  group_by(BCregion) |>
+#  summarise(area = sum(mountainArea)) |>
+#  ggplot(aes(x = fct_reorder(BCregion, area), y = area)) +
+#  geom_col() +
+#  labs(x = "Bioclimatic region",
+#       y = "Total mountain area (m2)") +
+#  coord_flip()+
+#  theme_bw()
```

Comment:
This code is commented out. Is it still needed?

Reply:
No, it's just for future reference. I added an annotation.